syntax = "proto3";

package chat;

message TextMessage {
  string sender = 1;
  string content = 2;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  string error_message = 2;
}

message RegisterRequest {
  string username = 1;
  string password = 2;
}

message RegisterResponse {
  bool success = 1;
  string error_message = 2;
}

message ListUsersRequest {
}

message ListUsersResponse {
  repeated string usernames = 1;
}